handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -1062,7 +1062,15 @@ private SSLEngineResult wrap(ByteBufAllocator alloc, SSLEngine engine, ByteBuf i
     public void channelInactive(ChannelHandlerContext ctx) throws Exception {
         boolean handshakeFailed = handshakePromise.cause() != null;
 
-        ClosedChannelException exception = new ClosedChannelException();
+        final Exception exception;
+        if (handshakePromise.isDone()) {
```

Comment:
There is a possibility that the channel will become inactive even before it was able to read anything and start the handshake. Consider `!isStateSet(STATE_HANDSHAKE_STARTED) || handshakePromise.isDone()`